src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +47,27 @@
     }
   }
 
-
+  /**
+   * Get the historical price for a specified Symbol for a certain range starting at date.
+   * @param symbol the symbol to get historical prices for.
+   * @param date the date to begin getting historical prices at.
+   * @param range the range to get historical prices at.
+   * @return a list of historical prices for the specified symbol and time frame.
+   */
+  public List<IexHistoricalPrice> getHistoricalPrices(
+      final String symbol,
+      final String range,
+      final String date) {
+    if (symbol.isEmpty()) {
+      return Collections.emptyList();
```

Comment:
Given that we expect a symbol in our endpoint, is emptyList the correct response here?
(not being snarky, genuine question)

Reply:
This was to catch the scenario where a request is made that has the symbol parameter, but the value is empty for example, this testcase:
```
        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
            .get("/iex/historicalPrice?symbol=") 
```
It would satisfy the required condition of the @RequestParam symbol parameter in the IexRestController, but would be unable to make a successful request to the IEX API as the symbol is a required parameter.

A few alternative solutions would be to make the if elses more explicit and to return an emptyList as a base return case. 

I could also make the symbol @RequestParam not required and test for null but I don't think that would be an effective work around.

Another option would be to do the check in the IexRestController but I think it would be best to keep the logic in the Service class.

Do you think I should still change it?

Reply:
Gotcha, that makes sense. In that case, you could look into adding request validation, because symbol= isn't really a valid request from the client. There are annotation based methods you could use to verify that symbol is populated.